Education/Visual Studio features.md

```diff
@@ -0,0 +1,71 @@
+
+
+
+### Extensions
+
+VS allows you to add extensions that enhance or alter functionalities provided by the IDE. There are a lot extensions that can be found at 'Extensions' tab or searched on the internet. Some of the extensions are free and some of them are paid. 
+
+Some free extensions that we suggest:
+
+- Fine Code Coverage - provides an overview of your test coverage giving information such as percentage of code covered, risky hotspot, covered branches etc.
+- Format document on Save - automatically applies default code formatting when saving a document.
+- Output enhancer - provides a better visual graphics when checking IDE console logs.
+- Visual Studio Spell Checker - adds spell checking for all files.
+
+
+
+### App settings configuration
+
+Different projects require different setting configurations, like connection string to database or URL to external API etc. These configurations in .NET are usually stored in appsettings.json file. Visual studio allows you to create different configurations through 'Configuration Manager' found in the 'Build' tab. By default projects are run in 'Debug' configuration with appsettings.json but it can be changed to use different settings, example appsettings.Test.json.
+
+Additionally VS allows you to use 'user secrets' by right clicking on project and clicking on 'Manage User Secrets'. These secrets are stored on your machine and cannot be pushed to github. These secrets override appsettings.json and are useful for local development.
+
+
+
+### Connecting to Azure 
+
+VS can be connected to Azure account by clicking on 'Manage connections' in 'Team' tab. When VS is connected to Azure it allows you a several different features, such as accessing DevOps git repository, selecting specific existing trigger (like blob or queue) from your Azure account etc. These Azure connections can also be used to connect to Azure database or other resources and manage them through VS itself.
+
+
+
+### Shortcuts
+
+VS provides shortcuts to some most common actions.  All shortcuts can be found by clicking on 'Options' under 'Tools' tab in the 'Keyboard' menu. Note that some extensions modify or add new shortcuts. Shortcuts can also be exported or imported.
+
+Some of the most common that we use:
+
+- `ctril+s` - to save a file.
+- `ctrl+e,d` - to format a file.
+- `ctrl+w` - to close a file (needs to be mapped in 'Keyboard' menu to command 'File.Close').
+- `ctrl+t` - to navigate through files, properties, methods etc.
+- `ctrl+g` - to navigate to specific line number in the file. 
+- `ctrl+f` - to search through files.
+- `ctrl+h` - open find and replace menu. 
+- `alt+enter` or `ctrl+.`- to open intelisense for the line you are at.
+- `ctrl+F12` - to navigate to method implementation or class definition.

```

Comment:
Here I would add differentiation between `F12` and `ctrl+F12`
- `F12` - navigate to the code definition
- `ctrl+F12` - navigate to the implementation or open list of implementations if there are more than one

Reply:
Added.